inference-engine/src/transformations/include/transformations/common_optimizations/convert_to_unsigned_nms_gather.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <transformations_visibility.hpp>
+#include <ngraph/pass/pass.hpp>
+
+namespace ngraph {
+namespace pass {
+
+class TRANSFORMATIONS_API ConvertToUnsignedNmsGather;
+
+}  // namespace pass
+}  // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief Converts Gather indices to unsigned if indices are from NMS 1st output:
+ */
+class ngraph::pass::ConvertToUnsignedNmsGather: public ngraph::pass::FunctionPass {
+public:
+    NGRAPH_RTTI_DECLARATION;
+    bool run_on_function(std::shared_ptr<ngraph::Function> f) override;
+};
+
+void convert_gather_indices_to_unsigned(const ngraph::NodeVector& gather_nodes);
+
+ngraph::NodeVector get_NmsGather_destinations(const std::shared_ptr<ngraph::Node>& nms_node);
```

Comment:
I don't think that we need this function in API. I propose to put it into `private` section or place it in `cpp` in anonymous namespace.